File: ohjelmointikielet/zoo-java/Shark.java

package fi.lab.zoo;

/**
 * @author Anni Karja
 */
public class Shark extends Animal {
    private String species;
    
    public Shark() {
        this("Timppa", new Date(), 120, "Great White");
    }
    
    public Shark(String name, Date birth, double weight, String species) {
        super(name, birth, weight);
        setSpecies(species);
    }
    
    @Override
    public String toString() {
        return super.toString() + " Species: " + getSpecies();
    }
    @Override
    public void move() {
        System.out.println("Shark " + getName() + " has moved!");
    }
    @Override
    public void utter() {
        System.out.println("Shark " + getName() + " goes brr!");
    }

    /**
     * @return the species
     */
    public String getSpecies() {
        return species;
    }

    /**
     * @param species the species to set
     */
    public void setSpecies(String species) {
        this.species = species;
    }
}

File: ohjelmointikielet/zoo-cs/Shark.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zoo
{
    class Shark : Animal
    {
        private string species;

        public string Species { get => species; set => species = value; }

        public Shark(string name = "Timppa", Date birth = null, double weight = 120, string species = "Great White")
            : base(name, birth, weight)
        {
            Species = species;
        }

        public override void move()
        {
            Console.WriteLine($"Shark {Name} has moved!");
        }

        public override void utter()
        {
            Console.WriteLine($"Shark {Name} goes brr!");
        }

        public override string ToString()
        {
            return $"{base.ToString()} Species: {Species}";
        }
    }
}
